### ISSUE TITLE
Voip push

### ISSUE THREAD

**Issue by krot3232** [esl/MongooseIM#2484]
MongooseIM version: 3.4.0
When sending stanza videocell:
`<jitsi xmlns = "urn: xmpp: jitsi" action = "newCall" participant = "1001@localhost" caller = "1145@localhost" room = "3B4E48FA-43DF-4857-BFDA-941A0AF5BAF8" hasVideo = "0" /> <body> Call with Name started at 11:15 </body>`
 need to send push to the mongooseIM_A server, but all other messages without xmlns = "urn: xmpp: jitsi" to the mongooseIM_B server.
One device has two different diviceid one for push voip, 2 for regular messages. For stanza videos, separate notifications. Is this a good idea?
Need to dynamically change the host push server, where in the code to fix it?

**Comment by fenek** [comment]
Hi @krot3232 

Let me rephrase what you've written, so we can confirm that I understand you properly.

You'd like to have two independent MIM servers (clusters) and the client would connect to both at the same time. Server A will be responsible for routing stanzas with `urn:xmpp:jitsi` namespace and server B for everything else.

Is it correct?
Also: what do you mean by `dynamically change the host push server`? Would you like to change it on the fly in the config? Or would you like to have some conditional logic to choose various push endpoints?

**Comment by krot3232** [comment]
One MongooseIM server.
User enabling notifications device id 1 (for all messages except video calls):
  <enable xmlns = "urn: xmpp: push: 0" jid = "pubsub.localhost"> <x xmlns = "jabber: x: data" type = "submit"> <field var = "device_id"> <value> 111 </value> </field> ..
then for video only:
  <enable xmlns = "urn: xmpp: push: 0" jid = "pubsub.localhost"> <x xmlns = "jabber: x: data" type = "submit"> <field var = "device_id"> <value> 222 </value> </field> ..
The user has 2 different device id (Apple Push Notification service SSL (production) && VoIP Services Certificate), depending on the purpose of the message.
When a message is sent to the user if it contains a video call (urn: xmpp: jitsi)
Need to change the endpoints push (mongoose_push_http, [{strategy, available_worker}], [{server, "serverA | serverB"}]})
forward messages to different mongoosePush api

**Comment by fenek** [comment]
I think that you need to change `src/mod_push_service_mongoosepush.erl:103` to pick different connection pool based on some information, possibly included in the form provided by the client. This alone should be sufficient. Well, perhaps some extra filtering in `src/event_pusher/mod_event_pusher_push_plugin_defaults.erl` could be useful as well.

**Comment by fenek** [comment]
@krot3232 

Do you need any further help with this issue? :)

**Comment by fenek** [comment]
Closing due to inactivity.
